src/lib/features/feature-toggle/feature-toggle-service.ts

```diff
@@ -358,6 +358,27 @@ class FeatureToggleService {
         }
     }
 
+    async validateStrategyLimit(
+        featureEnv: {
+            projectId: string;
+            environment: string;
+            featureName: string;
+        },
+        limit: number,
+    ) {
+        if (!this.flagResolver.isEnabled('resourceLimits')) return;
+        const existingCount = (
+            await this.featureStrategiesStore.getStrategiesForFeatureEnv(
```

Comment:
Originally I wanted to add a new count query but I decided to use the query for strategies and calculating length. In extreme scenarios the extra data we transfer from DB will be measured in 100s of kBs but this should not be an issue.